pkg/provisioners/gcppubsub/util/names.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+
+	eventduck "github.com/knative/eventing/pkg/apis/duck/v1alpha1"
+)
+
+// Underscore is used as a separator between the namespace and name so the concatenated version
+// can't accidentally overlap.
+
+// GenerateTopicName generates the GCP PubSub Topic name given the Knative Channel's namespace and
+// name.
+// Note that it must be stable. The controller and dispatcher rely on this being generated
+// identically.
+func GenerateTopicName(channelNamespace, channelName string) string {
```

Comment:
the kafka code does pass in a `Channel`, see:

```
func topicName(channel *eventingv1alpha1.Channel) string {
	return fmt.Sprintf("%s.%s", channel.Namespace, channel.Name)
}
```

Reply:
My preference is to get the strong type I can, but I use this function in a place that only has strings, so went this way. 

```golang
func (r *Receiver) sendEventToTopic(channel buses.ChannelReference, message *buses.Message) error {
	...
	topicName := util.GenerateTopicName(channel.Namespace, channel.Name)
        ...
```

Would you prefer I turn the strings into a real `Channel` instead?

Another option is to use the `buses.ChannelReference` type in this signature. Thoughts?

Reply:
@Harwayne I like your function - see here:

https://github.com/knative/eventing/pull/635

actually did also remove duplicate code in kafka provisioner already ;-)

Reply:
Switched to using the utils from #635.